Backend/src/resolverMap.ts

```diff
@@ -67,14 +92,18 @@ const resolverMap: IResolvers = {
       const username: string = input.username;
       const password: string = input.password;
       const isOAuth: boolean = false;
-      const user: IUser = {
-        username,
-        password,
-        email,
-        isOAuth,
-        refreshToken: null,
-      };
-      return userFacade.addNonOAuthUser(user);
+      if (username != "" && password != "" && email != "") {
+        let user: IUser = {
+          username,
+          password,
+          email,
+          isOAuth,
+          refreshToken: null,
+        };
+        return userFacade.addNonOAuthUser(user);
+      } else {
+        throw new UserInputError("Bad input");
```

Comment:
You could break this down even further for each input field if you wanted. Then use invalidArgs:
https://www.apollographql.com/docs/apollo-server/data/errors/#augmenting-error-details

```suggestion
        throw new UserInputError("Bad input", {
            invalidArgs: ["username", "password", "email"]
        });
```

Or, if checking a field at a time: 

```suggestion
        throw new UserInputError("Bad input", {
            invalidArgs: "username"
        });
```

Like here: 
![image](https://user-images.githubusercontent.com/11192971/83264892-11f12d80-a1c1-11ea-918c-514795584dde.png)